### PR TITLE
[SW-2195] Automatically generate Python confoguration from Scala code

### PR DESCRIPTION
**Pull request by jakubhava**

### CODE REVIEW

File: core/src/main/scala/ai/h2o/sparkling/backend/internal/InternalBackendConfUtils.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+       |  cloudName            : ${cloudName.getOrElse(
+      "Not set yet, it will be set automatically before starting H2OContext.")}
+       |  base port            : $basePort
+       |  cloudTimeout         : $cloudTimeout
+       |  log level            : $logLevel
+       |  nthreads             : $nthreads
+       |  drddMulFactor        : $drddMulFactor""".stripMargin
+
+  private[backend] override def getFileProperties: Seq[(String, _, _, _)] = super.getFileProperties :+ PROP_HDFS_CONF
+}
```
Comment (mn-mikke) [inline core/src/main/scala/ai/h2o/sparkling/backend/internal/InternalBackendConfUtils.scala:44]: nit: missing new line

File: core/src/main/scala/ai/h2o/sparkling/backend/external/ExternalBackendConf.scala [hunk 2]
```diff
@@ -244,16 +244,6 @@ trait ExternalBackendConf extends SharedBackendConf with Logging {
   def setExternalK8sServiceTimeout(timeout: Int): H2OConf = {
     set(PROP_EXTERNAL_K8S_SERVICE_TIMEOUT._1, timeout.toString)
   }
-
-  def externalConfString: String =
```
Comment (mn-mikke) [inline core/src/main/scala/ai/h2o/sparkling/backend/external/ExternalBackendConf.scala:248]: why this was moved to a dedicated trait?

File: core/src/main/scala/ai/h2o/sparkling/backend/external/ExternalBackendConfUtils.scala [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+       |  backend cluster mode : $backendClusterMode
+       |  cluster start mode   : $clusterStartMode
+       |  cloudName            : ${cloudName.getOrElse("Not set yet")}
+       |  cloud representative : ${h2oCluster.getOrElse("Not set, using cloud name only")}
+       |  base port            : $basePort
+       |  log level            : $logLevel
+       |  nthreads             : $nthreads""".stripMargin
+}
+
+
```
Comment (mn-mikke) [inline core/src/main/scala/ai/h2o/sparkling/backend/external/ExternalBackendConfUtils.scala:41]: nit: extra line

File: core/src/main/scala/ai/h2o/sparkling/backend/internal/InternalBackendConf.scala [hunk 4]
```diff
@@ -83,20 +83,6 @@ trait InternalBackendConf extends SharedBackendConf {
   }
 
   def setSpreadRddRetriesTimeout(timeout: Int): H2OConf = set(PROP_SPREADRDD_RETRIES_TIMEOUT._1, timeout.toString)
-
-  def internalConfString: String =
```
Comment (mn-mikke) [inline core/src/main/scala/ai/h2o/sparkling/backend/internal/InternalBackendConf.scala:87]: why this was moved to a dedicated trait?

File: core/src/main/scala/ai/h2o/sparkling/backend/external/ExternalBackendConfUtils.scala [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+package ai.h2o.sparkling.backend.external
+
+import ai.h2o.sparkling.H2OConf
+import ai.h2o.sparkling.backend.SharedBackendConf
+import org.apache.spark.expose.Logging
+
+/**
+  * External backend configuration
+  */
+trait ExternalBackendConfUtils extends SharedBackendConf with Logging {
```
Comment (mn-mikke) [inline core/src/main/scala/ai/h2o/sparkling/backend/external/ExternalBackendConfUtils.scala:27]: nit: Can you think of a more precise name than `*Utils`?

Comment (jakubhava) [reply core/src/main/scala/ai/h2o/sparkling/backend/external/ExternalBackendConfUtils.scala:27]: Yup, what about `*Extensions`?

File: core/src/main/scala/ai/h2o/sparkling/backend/internal/InternalBackendConf.scala (hunk 4 above)
Comment (jakubhava) [reply core/src/main/scala/ai/h2o/sparkling/backend/internal/InternalBackendConf.scala:87]: We get all the methods defined in the *Conf classes and automatically generated getters & setters. This method is not supposed to be part of the generated configuration

File: core/src/main/scala/ai/h2o/sparkling/backend/internal/InternalBackendConfUtils.scala (hunk 1 above)
Comment (jakubhava) [reply core/src/main/scala/ai/h2o/sparkling/backend/internal/InternalBackendConfUtils.scala:44]: 👍

File: core/src/main/scala/ai/h2o/sparkling/backend/external/ExternalBackendConfUtils.scala (hunk 3 above)
Comment (jakubhava) [reply core/src/main/scala/ai/h2o/sparkling/backend/external/ExternalBackendConfUtils.scala]: 👍

File: core/src/main/scala/ai/h2o/sparkling/backend/internal/InternalBackendConf.scala (hunk 4 above)
Comment (mn-mikke) [reply core/src/main/scala/ai/h2o/sparkling/backend/internal/InternalBackendConf.scala:87]: I see, thanks

File: core/src/main/scala/ai/h2o/sparkling/backend/external/ExternalBackendConfUtils.scala (hunk 5 above)
Comment (mn-mikke) [reply core/src/main/scala/ai/h2o/sparkling/backend/external/ExternalBackendConfUtils.scala]: SGTM

**Pull request merged**